pkgs/shells/fish/default.nix

```diff
@@ -35,28 +35,56 @@ let
     #     source both, but source the more global configuration files earlier
     #     than the more local ones, so that more local configurations inherit
     #     from but override the more global locations.
-    
+
     if test -f /etc/fish/config.fish
       source /etc/fish/config.fish
     end
-    
+
     #                                                                             #
     ############### ↑ Nix hook for sourcing /etc/fish/config.fish ↑ ###############
   '';
 
+
+  nixEnvironmentSetupScriptPrefix = "@NIX_STORE@/../var/nix/profiles/default/etc/profile.d";
+  selectPosixPreInitScript = if hostPlatform.isDarwin then ''
+    # anchor for indentation
```

Comment:
That’s brittle. Why not `writeText` and `source`?

Reply:
> That’s brittle. Why not writeText and source?

You mean specifically the indentation hack? Because it works, the stakes are low (the whitespace here won't break anything), and it seems good to avoid sourcing additional files during shell startup if possible. It's also more readable.

I can let that go, though.

Reply:
If you want to avoid opening files, I’d hate to tell you how many unsuccessful stats most of the software we use do, searching for files in locations that are just crazy by nix standards …